documentation/content/operators-manual/resilience-data-redundancy.md

```diff
@@ -0,0 +1,126 @@
+
+Each Wallaroo worker must be compiled with the `resilience=on` option and must use the `--run-with-resilience` flag, the `--resilience-enable-io-journal` flag, and the `--resilience-dos-server hostnameOrIpAddress:9999` flag.
+
+## Asynchronous journal file replication
+
+Wallaroo's I/O journal file replication uses the primary/secondary replication technique: the worker's local journal file is always the primary copy, and all remote DOS server replicas are secondary copies.  Each secondary copy is replicated asynchronously.  When the TCP connection to a remote DOS server is interrupted, a new connection is automatically reestablished when feasible.  Any data written to the primary file while disconnected from a remote DOS server is copied after a new connection is made.
+
+If more than one remote DOS servers are used by a single Wallaroo worker, then it is quite likely that the secondary copies will not be exactly the same size.  All secondary copies will lag behind the primary copy by some number of milliseconds or even full seconds.
+
+We know that Wallaroo a worker creates I/O journal files in strictly append-only fashion.  This append-only property makes it easy to calculate which secondary copy is the most recent or "most up-to-date" copy: the largest copy (i.e., total file size) is the most recent copy.
+
+## Running a DOS file server
+
+The DOS server is run with a single mandatory command line argument: a path to the local file system where the DOS server's files will be stored.  The server listens to TCP port 9999 on all network interfaces.
+
+Example command line usage is:
+
+    mkdir -p /var/lib/dos-server
+    testing/tools/dos-dumb-object-service/dos-server.py /var/lib/dos-server
+
+A DOS server may be used by multiple Wallaroo workers.  Each worker will store its I/O journal files in a separate subdirectory.  For example, the `initializer` worker's I/O journal files will be stored in the `/var/lib/dos-server/initializer` subdirectory.
+
+## Running Wallaroo to use a remote DOS file server (thus enabling data redundancy)
+
+NOTE: The Wallaroo executable (e.g., `machida` or `machida3`) compiled with the `resilience=on` option.
+
+The following command line flags are recommended to enable data redundancy.
+
+1. `--run-with-resilience` to enable crash & restart resiliency.
+
+2. `--resilience-dir /path/to/dir` to specify where the local I/O journal files will be stored.
+
+3. `--resilience-enable-io-journal` to enable I/O journalling of Wallaroo's crash resilience data.
+
+4. `--resilience-no-local-file-io`, which disables the creation of non-I/O journal files.
+
+5. `--dos-server host:9999` to specify where the DOS server can be contacted.  The `host` string may be a DNS hostname or IPv4 address.
```

Comment:
@slfritchie shouldn't this be `--resilience-dos-server` instead?

Reply:
nice catch

Reply:
Indeed, nice catch.  Fixed by c94241d